plot-lib/plot/private/plot2d/rectangle.rkt

```diff
@@ -59,24 +62,9 @@
     [(or (> alpha 1) (not (rational? alpha)))  (fail/kw "real in [0,1]" '#:alpha alpha)]
     [else
      (let ([rects  (sequence->listof-vector 'rectangles rects 2)])
-       (match-define (list (vector (ivl #{x1s : (Listof (U Real #f))}
-                                        #{x2s : (Listof (U Real #f))})
-                                   (ivl #{y1s : (Listof (U Real #f))}
-                                        #{y2s : (Listof (U Real #f))}))
-                           ...)
-         rects)
-       (define rxs (filter rational? (append x1s x2s)))
-       (define rys (filter rational? (append y1s y2s)))
-       (cond
-         [(or (empty? rxs) (empty? rys))  (renderer2d #f #f #f #f)]
-         [else
-          (let ([x-min  (if x-min x-min (apply min* rxs))]
-                [x-max  (if x-max x-max (apply max* rxs))]
-                [y-min  (if y-min y-min (apply min* rys))]
-                [y-max  (if y-max y-max (apply max* rys))])
-            (renderer2d (vector (ivl x-min x-max) (ivl y-min y-max)) #f default-ticks-fun
-                        (rectangles-render-proc rects color style line-color line-width line-style
-                                                alpha label)))]))]))
```

Comment:
Removing this code breaks the example in the docs:

```
#lang racket/base
(require plot)
(plot (rectangles (list (vector (ivl -1 0) (ivl -1 1))
                        (vector (ivl 0 2) (ivl 1 2)))))
```

because plot is no longer able to infer the bounds.

- - -

I think the old code would be able to handle open-ended intervals if the condition on L71 was something like:

```
[(or (and (empty? x1s) (empty? x2s)) (and (empty? y1s) (empty? y2s)))  (renderer2d #f #f #f #f)]
```

Or maybe, this line can just be removed.